### PR TITLE
Input whole video

### PR DESCRIPTION
**Pull request by morten-j**

### CODE REVIEW

File: sanic/src/server.py [hunk 1]
```diff
@@ -40,7 +40,6 @@ async def hands(request):
         content:
             application/json: {}
     """
-    
     flipped = request.args.get("flip")
```
Comment (CasperKjaerhus) [inline sanic/src/server.py:43]: ka vi lige få den linje tilbage? :)

**Review by mKiddeT — APPROVED** [review]
🤔 hmm

File: sanic/src/train.py [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+    with open(pkl_filename2, 'rb') as file:
+        train_labels = pickle.load(file)
+
+    pkl_filename3 = "./data/testdata.pkl"
+    with open(pkl_filename3, 'rb') as file:
+        test_data = pickle.load(file)
+
+    pkl_filename4 = "./data/testlabels.pkl"
+    with open(pkl_filename4, 'rb') as file:
+        test_labels = pickle.load(file)
```
Comment (CasperKjaerhus) [inline sanic/src/train.py]: put it down, kill it.

Comment (CasperKjaerhus) [inline sanic/src/train.py]:
```suggestion
if args["load"] == "True":

    with open("./data/traindata.pkl", 'rb') as file:
        train_data = pickle.load(file)

    with open("./data/trainlabels.pkl", 'rb') as file:
        train_labels = pickle.load(file)
        
    with open("./data/testdata.pkl", 'rb') as file:
        test_data = pickle.load(file)

    with open("./data/testlabels.pkl", 'rb') as file:
        test_labels = pickle.load(file)
```

Comment (mKiddeT) [reply sanic/src/train.py]: beta cuck

Comment (mKiddeT) [reply sanic/src/train.py]: beta cuck

File: sanic/src/train.py [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+    with open(pkl_filename2, 'wb') as file:
+            pickle.dump(train_labels, file)
+
+    pkl_filename3 = "./data/testdata.pkl"
+    with open(pkl_filename3, 'wb') as file:
+            pickle.dump(test_data, file)
+
+    pkl_filename4 = "./data/testlabels.pkl"
+    with open(pkl_filename4, 'wb') as file:
+            pickle.dump(test_labels, file)
```
Comment (CasperKjaerhus) [inline sanic/src/train.py]:
```suggestion
    with open("./data/traindata.pkl", 'wb') as file:
            pickle.dump(train_data, file)

    with open("./data/trainlabels.pkl", 'wb') as file:
            pickle.dump(train_labels, file)

    with open("./data/testdata.pkl", 'wb') as file:
            pickle.dump(test_data, file)

    with open("./data/testlabels.pkl", 'wb') as file:
            pickle.dump(test_labels, file)
```

File: sanic/src/train.py [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+else:
+    print("Specify valid load argument!!!")
```
Comment (CasperKjaerhus) [inline sanic/src/train.py:106]: why so mad

File: sanic/src/train.py [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+mask_input = keras.Input((MAX_SEQ_LENGTH,), dtype="bool")
+
+# Refer to the following tutorial to understand the significance of using `mask`:
+# https://keras.io/api/layers/recurrent_layers/gru/
+x = keras.layers.GRU(524, return_sequences=True)(
+    frame_features_input, mask=mask_input
+)
+x = keras.layers.GRU(124)(x)
+x = keras.layers.Dropout(0.4)(x)
+x = keras.layers.Dense(64, activation="relu")(x)
```
Comment (CasperKjaerhus) [inline sanic/src/train.py:124]: kan du lige forklare den her for min not-working python brian?

File: sanic/src/utils.py [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+    try:
+        while True:
+            ret, frame = cap.read()
+            if not ret: # Pad data
+                for i in range(len(frames), MAX_SEQ_LENGTH):
+                    frames.append(pad_frame)
+                break
+            frame = crop_center_square(frame)
+            frame = cv2.resize(frame, resize)
+            frame = frame[:, :, [2, 1, 0]]
```
Comment (CasperKjaerhus) [inline sanic/src/utils.py]:
```suggestion
            frame = frame[:, :, [2, 1, 0]] # Converts frame from BGR to RGB
```

**Review by CasperKjaerhus — APPROVED** [review]
should be refactored in future

**Pull request merged**